frontend/src/components/ui/blockEditor/SpecsInterface.jsx

```diff
@@ -0,0 +1,195 @@
+        return; // Skip these keys
+      }
+  
+      const newPath = path.concat(key);
+      const newKey = prefix ? `${prefix}.${key}` : key;
+      if (typeof data[key] === 'object' && !Array.isArray(data[key]) && data[key] !== null) {
+        // Recursively call flattenSpecs for nested objects
+        fields = fields.concat(flattenSpecs(data[key], newKey, newPath));
+      } else {
+        // Construct label based on the path
+        let label = key; // Default label
+        if (newPath.includes('inputs') || newPath.includes('outputs')) {
+          const typeLabel = newPath[0].charAt(0).toUpperCase() + newPath[0].slice(1); // Capitalize first letter
+          const varName = newPath[newPath.length - 2]; // Get variable name from the second last item in the path
+          label = `${typeLabel} ${varName} Type`; // Correctly format label
+        }
+        fields.push({ key: newKey, label, value: data[key] });
+      }
+    });
+    return fields;
+  };
+
+  const handleInputChange = (field, value) => {
+    // Update the value in the flattened specs array
+    const newFlattenedSpecs = flattenedSpecs.map(item => {
+      if (item.key === field) {
+        return { ...item, value: value };
+      }
+      return item;
+    });
+    setFlattenedSpecs(newFlattenedSpecs);
+
+    // Update the original specs object
+    const keys = field.split('.');
+    const lastKey = keys.pop();
+    const lastObj = keys.reduce((obj, key) => obj[key] = obj[key] || {}, specs);
+    lastObj[lastKey] = value;
+    setSpecs({ ...specs });
+  };
+
+  function specsValuesUpdate(blockFolderName, specs, pipelineData) {
+    // Clone the relevant block specs from pipeline data
+    const pipelineBlockSpecs = JSON.parse(JSON.stringify(pipelineData[blockFolderName]));
+  
+    // Define a set of keys to skip
+    const keysToSkip = new Set(['connections', 'relays', 'command_line', 'pos_x', 'pos_y', 'pos_z', 'parameters']);
+  
+    // Function to recursively update the values based on specific rules
+    function updateValues(target, source, path = []) {
+      Object.keys(source).forEach(key => {
+        // Skip updates for specific keys
+        if (keysToSkip.has(key)) {
+          return;
+        }
+  
+        const newPath = path.concat(key);
+        if (typeof source[key] === 'object' && !Array.isArray(source[key]) && source[key] !== null) {
+          // Ensure the target path exists or create it
+          if (!target[key]) target[key] = {};
+  
+          // Continue recursion
+          updateValues(target[key], source[key], newPath);
+        } else {
+          // Update the leaf nodes
+          target[key] = source[key];
+        }
+      });
+    }
+  
+    // Start the update process
+    updateValues(pipelineBlockSpecs, specs);
+  
+    return pipelineBlockSpecs;
+  } 
+  
+  const saveSpecs = () => {
+    setIsLoading(true);
+    fetch(`${serverAddress}/update-specs`, {
```

Comment:
Don't need to modify specs with a new api call -- just updating the graph will sync to disk